### DISCUSSION TITLE
HAVING ISSUE

### DISCUSSION THREAD

**Discussion by ORV50** [jameszah/ESP32-CAM-Video-Telegram#21] [Show and tell]
Please help me to fix this issue when uploading the code:
Arduino: 1.8.19 (Windows 10), Board: "ESP32 Wrover Module, Huge APP (3MB No OTA/1MB SPIFFS), QIO, 80MHz, 921600, None"

C:\Users\oscar\Documents\Arduino\TELEGRAM_VIDEO_OSCAR_CAM_MAYO\TELEGRAM_VIDEO_OSCAR_CAM_MAYO.ino: In function 'void handleNewMessages(int)':

TELEGRAM_VIDEO_OSCAR_CAM_MAYO:389:27: error: 'class UniversalTelegramBot' has no member named 'sendMultipartFormDataToTelegramWithCaption'

         String sent = bot.sendMultipartFormDataToTelegramWithCaption("sendPhoto", "photo", "img.jpg",

                           ^

TELEGRAM_VIDEO_OSCAR_CAM_MAYO:401:47: error: no matching function for call to 'UniversalTelegramBot::sendPhotoByBinary(String&, const char [11], size_t&, bool (&)(), uint8_t (&)(), std::nullptr_t, std::nullptr_t)'

                               nullptr, nullptr);

                                               ^

In file included from C:\Users\oscar\Documents\Arduino\TELEGRAM_VIDEO_OSCAR_CAM_MAYO\TELEGRAM_VIDEO_OSCAR_CAM_MAYO.ino:2:0:

C:\Users\oscar\Documents\Arduino\libraries\UniversalTelegramBot\src/UniversalTelegramBot.h:76:10: note: candidate: String UniversalTelegramBot::sendPhotoByBinary(String, String, int, MoreDataAvailable, GetNextByte)

   String sendPhotoByBinary(String chat_id, String contentType, int fileSize,

          ^

C:\Users\oscar\Documents\Arduino\libraries\UniversalTelegramBot\src/UniversalTelegramBot.h:76:10: note:   candidate expects 5 arguments, 7 provided

TELEGRAM_VIDEO_OSCAR_CAM_MAYO:435:45: error: no matching function for call to 'UniversalTelegramBot::sendPhotoByBinary(String&, const char [11], size_t&, bool (&)(), uint8_t (&)(), std::nullptr_t, std::nullptr_t)'

                             nullptr, nullptr);

                                             ^

In file included from C:\Users\oscar\Documents\Arduino\TELEGRAM_VIDEO_OSCAR_CAM_MAYO\TELEGRAM_VIDEO_OSCAR_CAM_MAYO.ino:2:0:

C:\Users\oscar\Documents\Arduino\libraries\UniversalTelegramBot\src/UniversalTelegramBot.h:76:10: note: candidate: String UniversalTelegramBot::sendPhotoByBinary(String, String, int, MoreDataAvailable, GetNextByte)

   String sendPhotoByBinary(String chat_id, String contentType, int fileSize,

          ^

C:\Users\oscar\Documents\Arduino\libraries\UniversalTelegramBot\src/UniversalTelegramBot.h:76:10: note:   candidate expects 5 arguments, 7 provided

C:\Users\oscar\Documents\Arduino\TELEGRAM_VIDEO_OSCAR_CAM_MAYO\TELEGRAM_VIDEO_OSCAR_CAM_MAYO.ino: In function 'void send_the_picture()':

TELEGRAM_VIDEO_OSCAR_CAM_MAYO:1121:23: error: 'class UniversalTelegramBot' has no member named 'sendMultipartFormDataToTelegramWithCaption'

     String sent = bot.sendMultipartFormDataToTelegramWithCaption("sendPhoto", "photo", "img.jpg",

                       ^

TELEGRAM_VIDEO_OSCAR_CAM_MAYO:1125:23: error: 'class UniversalTelegramBot' has no member named 'sendMultipartFormDataToTelegramWithCaption'

     String sent = bot.sendMultipartFormDataToTelegramWithCaption("sendPhoto", "photo", "img.jpg",

                       ^

C:\Users\oscar\Documents\Arduino\TELEGRAM_VIDEO_OSCAR_CAM_MAYO\TELEGRAM_VIDEO_OSCAR_CAM_MAYO.ino: In function 'void send_the_video()':

TELEGRAM_VIDEO_OSCAR_CAM_MAYO:1144:22: error: 'class UniversalTelegramBot' has no member named 'sendMultipartFormDataToTelegramWithCaption'

   String sent2 = bot.sendMultipartFormDataToTelegramWithCaption("sendDocument", "document", strftime_buf,

                      ^

Multiple libraries were found for "UniversalTelegramBot.h"

 Used: C:\Users\oscar\Documents\Arduino\libraries\UniversalTelegramBot

 Not used: C:\Program Files (x86)\Arduino\libraries\Universal-Arduino-Telegram-Bot-master

 Not used: C:\Users\oscar\Documents\Arduino\libraries\Universal-Arduino-Telegram-Bot-master

 Not used: C:\Users\oscar\Documents\Arduino\libraries\Universal-Arduino-Telegram-Bot-1.3.0

Multiple libraries were found for "WiFi.h"

 Used: C:\Users\oscar\AppData\Local\Arduino15\packages\esp32\hardware\esp32\1.0.5\libraries\WiFi

 Not used: C:\Program Files (x86)\Arduino\libraries\WiFi

exit status 1

'class UniversalTelegramBot' has no member named 'sendMultipartFormDataToTelegramWithCaption'
Please help I have found this project very useful but I can't continue.
Thank you.
Oscar
oscarrv13@gmail.com

**Comment by jameszah** [comment]
You need these 3 files:

![image](https://user-images.githubusercontent.com/36938190/171443338-cfc8b707-b2cf-4409-90fe-b772f531f2e7.png)

And you have to drag them all into the Arduino IDE, like this:

![image](https://user-images.githubusercontent.com/36938190/171443540-705cb547-7ff5-4e01-aeb6-f1c7827aaf5b.png)
